### PR TITLE
[chainerx] Adds `logical_xor`

### PR DESCRIPTION
**Pull request by ishanrai05**
Solves part of issue #6423

### CODE REVIEW

File: tests/chainerx_tests/unit_tests/routines_tests/test_logic.py [hunk 1]
```diff
@@ -225,12 +225,14 @@ def logical_and(xp, a, b):
 def logical_or(xp, a, b):
     return xp.logical_or(a, b)
 
+def logical_xor(xp, a, b):
+    return xp.logical_or(a, b)
```
Comment (hvy) [inline tests/chainerx_tests/unit_tests/routines_tests/test_logic.py]: Hm, you're testing the wrong routine here.
```suggestion
    return xp.logical_xor(a, b)
```

**Comment by hvy** [comment]
Thanks for the PR. Could you take a look at my comment above? It doesn't look your implementation is tested.

File: tests/chainerx_tests/unit_tests/routines_tests/test_logic.py (hunk 1 above)
Comment (ishanrai05) [reply tests/chainerx_tests/unit_tests/routines_tests/test_logic.py]: oops.

**Comment by ishanrai05** [comment]
@hvy Fixed it. Can you please take a look again?

**Comment by hvy** [comment]
Can you take a look at the Travis failures? There are some style errors in your Python and C++ code. 

Maybe you're aware of this already but once a change is pushed to a PR branch, a new Travis build is queued/triggered. We need to make sure that these tests pass before merging.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
LGTM!

**Pull request merged**